src/incremental/compareCIL.ml

```diff
@@ -100,10 +156,22 @@ let compareCilFiles ?(eq=eq_glob) (oldAST: file) (newAST: file) =
   (* Store a map from functionNames in the old file to the function definition*)
   let oldMap = Cil.foldGlobals oldAST addGlobal GlobalMap.empty in
   let newMap = Cil.foldGlobals newAST addGlobal GlobalMap.empty in
+
+  let global_rename_mapping: method_rename_assumptions = Cil.foldGlobals newAST (fun (current_global_rename_mapping: method_rename_assumption list) global ->
+    match generate_global_rename_mapping oldMap global with
+      | Some rename_mapping -> current_global_rename_mapping @ [rename_mapping]
```

Comment:
Appending to a list in `foldGlobals` is stupidly inefficient, especially if there are many. Just prepend instead. And if you really need them in the right order, just reverse the list at the end.

Reply:
Thank you I implemented that optimization.